test/Servant/QuickCheck/InternalSpec.hs

```diff
@@ -142,11 +145,29 @@ queryFlagsSpec = describe "QueryFlags" $ do
   it "reduce to an HTTP query string correctly" $ do
     let rng = mkQCGen 0
         burl = BaseUrl Http "localhost" 80 ""
-        gen = genRequest flagsAPI
+        gen = runGenRequest flagsAPI
         req = (unGen gen rng 0) burl
         qs = C.unpack $ queryString req
     qs `shouldBe` "one&two"
 
+makeRandomRequest :: Proxy LargeAPI -> BaseUrl -> IO Integer
+makeRandomRequest large burl = do
+  req <- generate $ runGenRequest large
+  pure $ fst . fromJust . C.readInteger . C.drop 1 . path $ req burl
+
+
+unbiasedGenerationSpec :: Spec
+unbiasedGenerationSpec = describe "Unbiased Generation of requests" $
+
+  it "frequency paired with generated endpoint should be more randomly distributed" $ do
+    let burl = BaseUrl Http "localhost" 80 ""
+    let runs = 10000 :: Double
+    someRequests <- replicateM 10000 (makeRandomRequest largeApi burl)
+    let mean = (sum $ map fromIntegral someRequests) / runs
+    let variancer x = let ix = fromIntegral x in (ix - mean) * (ix - mean)
+    let variance = (sum $ map variancer someRequests) / runs - 1
+    (variance > 20) `shouldBe` True
```

Comment:
So, this is likely to be a flaky test, but I wasn't sure what a good statistical test for a random distribution here would be. I first thought of variance, but then kind of guessed what a good variance metric would be. 

Does anyone have any good suggestions for testing the uniformity of the distribution of `someRequests` (which is just a giant list of numbers produced by the `HasGenRequest` instance for APIs.